compiler/src/middle_end/linearize.re

```diff
@@ -757,35 +757,56 @@ let rec transl_imm =
         ),
       ],
     );
-  | TExpRecord(args) =>
+  | TExpRecord(base, args) =>
+    let base_imm = Option.map(transl_imm, base);
     let tmp = gensym("record");
-    let definitions =
-      Array.to_list @@ Array.map(((desc, def)) => def, args);
-    let definitions =
-      List.map(
-        fun
-        | Kept(_) => assert(false)
-        | Overridden(name, def) => (name, def),
-        definitions,
-      );
     let (new_args, new_setup) =
       List.split(
         List.map(
-          (({txt: name, loc}, expr)) => {
-            let (var, setup) = transl_imm(expr);
-            (
-              (Location.mkloc(Identifier.string_of_ident(name), loc), var),
-              setup,
-            );
-          },
-          definitions,
+          fun
+          | (ld, Kept) => {
```

Comment:
Generally, I try to avoid the `fun` keyword because it isn't friendly for new folks jumping into the compiler.

Reply:
It appears that `fun` is used in a bunch of other places in the compiler; should an issue be made to remove them all?

Reply:
> should an issue be made to remove them all?

Probably not, we can clean them up as encountered (I'm also guessing a lot of them are from the ocaml source code).